metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -117,6 +117,104 @@ case class Indexer(indexProviders: IndexProviders)(implicit rc: ReportContext) {
           importedBuild.dependencyModules
         )
 
+        // hack for mill build sources until a proper BSP wrapped sources mapping gets into
+        // the official protocol
```

Comment:
@tgodzik I managed to recreate `MappedSource`s without BSP. The values seem to be set correctly, but still metals doesn't work correctly...
I think I need some help.

Reply:
I will take a look!